extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,47 +20,170 @@ export abstract class ToolBase implements ITool {
-	protected abstract readonly versionCommand: string;
+
+	protected abstract readonly versionCommand: Command;
+
+	protected async getInstallationPath(): Promise<string | undefined> {
+		return await Promise.resolve(undefined);
+	}
+
+	protected get installationSearchPaths(): (string | undefined)[] {
+		return [this.storagePath];
+	}
+
+	protected get downloadPath(): string {
+		return this.storagePath;
+	}
+
+	protected logToOutputChannel(data: string | Buffer, header?: string): void {
+		this._platformService.logToOutputChannel(data, header);
+	}
+
+	public get status(): ToolStatus {
+		return this._status;
+	}
+
+	public get storagePath(): string {
+		return this._platformService.storagePath();
+	}
+
+	public get osType(): OsType {
+		return this._osType;
+	}
 
 	public get version(): SemVer | undefined {
 		return this._version;
 	}
 
-	public get isInstalled(): boolean {
-		return this._isInstalled;
+	public get fullVersion(): string | undefined {
+		return this._version && this._version.version;
+
 	}
 
 	public get statusDescription(): string | undefined {
 		return this._statusDescription;
 	}
 
-	public loadInformation(): Promise<void> {
-		if (this._isInstalled) {
-			return Promise.resolve();
+	protected get installationCommands(): Command[] {
+		return this.allInstallationCommands[this.osType];
+	}
+
+	public getErrorMessage(error: any): string {
+		return this._platformService.getErrorMessage(error);
+	}
+
+	protected async getPip3InstallLocation(packageName: string): Promise<string> {
+		const command = `pip3 show ${packageName}`;
+		const pip3ShowOutput: string = await this._platformService.runCommand(command, /* options */ undefined, /*sudo?*/ false, /*commandString*/ undefined, /*ignoreError*/ true);
+		const installLocation = /^Location\: (.*)$/gim.exec(pip3ShowOutput);
+		let retValue = installLocation && installLocation[1];
+		if (retValue === undefined || retValue === null) {
+			this.logToOutputChannel(`   >${command}`);
+			this.logToOutputChannel(`   Could not find 'Location' in the output:`);
+			this.logToOutputChannel(pip3ShowOutput, `   output:`);
+			return '';
+		} else {
+			return retValue;
 		}
-		this._isInstalled = false;
-		this._statusDescription = undefined;
-		this._version = undefined;
-		this._versionOutput = undefined;
-		return this._platformService.runCommand(this.versionCommand).then((stdout) => {
-			this._versionOutput = stdout;
-			this._version = this.getVersionFromOutput(stdout);
-			if (this._version) {
-				this._isInstalled = true;
-			} else {
-				throw localize('deployCluster.InvalidToolVersionOutput', "Invalid output received.");
+	}
+
+	public get outputChannelName(): string {
+		return this._platformService.outputChannelName();
+	}
+
+	public showOutputChannel(preserveFocus?: boolean | undefined): void {
+		this._platformService.showOutputChannel(preserveFocus);
+	}
+
+	public async install(updateDisplayTableData: (tool: ITool) => void, thisObject: any): Promise<void> {
+		try {
```

Comment:
Why not just pass in the dialog object to this call and then call dialog.updateDisplayTableData directly? That seems much simpler than this complex back and forth

Reply:
yeah agreed. Will change this.

Reply:
fixed.